### PR TITLE
Shuffleboard Layouts

### PR DESCRIPTION
**Pull request by Martysh12**
Display helpful information in Shuffleboard for robot operators and debugging.

### CODE REVIEW

**Comment by Martysh12** [comment]
image
image

**Comment by Martysh12** [comment]
image

**Comment by Martysh12** [comment]
To-do list (for myself):
 - [x] Make a tab that condenses all the information, to be used during matches by the robot operators.
 - [x] ~~Convert shooter speed values from RPM to m/s (AKA the expected exit speed of the note out of the shooter).~~ Won't do

File: src/main/java/frc/robot/subsystems/DriveSubsystem.java [hunk 1]
```diff
@@ -106,20 +194,18 @@ private void updateOdometry() {
...
-    public void arcadeDrive(double linearSpeed, double rotationSpeed){
-        differentialDrive.arcadeDrive(linearSpeed, rotationSpeed);
-    }
-
+    @Override
     public void periodic() {
         updateOdometry();
+        updateShuffleboard();
 
         SmartDashboard.putNumber("angleOfBot", getAngleDegrees());
```
Comment (tomtzook) [inline src/main/java/frc/robot/subsystems/DriveSubsystem.java]: why do we need all these SmartDashboard writes if you have the new shuffleboard entries approach?

File: src/main/java/frc/robot/subsystems/DriveSubsystem.java [hunk 2]
```diff
@@ -42,53 +54,129 @@ public DriveSubsystem() {
...
                 getRightDistancePassedMeters()
         );
 
-
         initialize();
+        setUpShuffleboard();
+    }
 
-
+    private void setUpShuffleboard() {
```
Comment (tomtzook) [inline src/main/java/frc/robot/subsystems/DriveSubsystem.java:118]: put `private` methods below all the `public` ones for organization.

File: src/main/java/frc/robot/subsystems/ShooterSystem.java [hunk 3]
```diff
@@ -32,11 +43,6 @@ public ShooterSystem() {
         motorLB.restoreFactoryDefaults();
         motorRB.restoreFactoryDefaults();
 
-        motorRT.setIdleMode(CANSparkBase.IdleMode.kBrake);
```
Comment (tomtzook) [inline src/main/java/frc/robot/subsystems/ShooterSystem.java]: don't add changes that are not relevant to the branch. You see a problem: report and fix in a separate branch

File: src/main/java/frc/robot/subsystems/ArmSystem.java [hunk 4]
```diff
@@ -23,17 +32,33 @@ public ArmSystem() {
...
+                .withPosition(0, 0);
     }
 
     public boolean reachedATargetAngle(double targetAngle) {
         return MathUtil.isNear(targetAngle, getAbsEncoderPositionDegrees(), NEAR_ANGLE_TOLERANCE);
     }
 
-    public void moveToAngle(double output) {
-        if ((output < 0 && getAbsEncoderPositionDegrees() > RobotMap.ARM_MAX_ANGLE) || output > 0 && getAbsEncoderPositionDegrees() < RobotMap.ARM_MIN_ANGLE){
+    public void move(double power) {
```
Comment (tomtzook) [inline src/main/java/frc/robot/subsystems/ArmSystem.java]: again, don't make changes not relevant to your branch. Especially naming changes.

We are working in a team, you do not have free rein to change things at will

File: src/main/java/frc/robot/commands/ArmMoveToFloorCommand.java [hunk 5]
```diff
@@ -23,9 +23,9 @@ public void initialize() {
 
     @Override
     public void execute() {
-        double output = pidController.calculate(armSystem.getAbsEncoderPositionDegrees(), RobotMap.ARM_FLOOR_ANGLE);
+        double power = pidController.calculate(armSystem.getAbsEncoderPositionDegrees(), RobotMap.ARM_FLOOR_ANGLE);
```
Comment (tomtzook) [inline src/main/java/frc/robot/commands/ArmMoveToFloorCommand.java]: No, not relevant to the branch and unnecessary

File: src/main/java/frc/robot/commands/ArmMoveToShooterCommand.java [hunk 6]
```diff
@@ -25,7 +25,7 @@ public void initialize() {
     public void execute() {
         double output = pidController.calculate(armSystem.getAbsEncoderPositionDegrees(),  RobotMap.ARM_SHOOTER_ANGLE);
 
-        armSystem.moveToAngle(output);
+        armSystem.move(output);
```
Comment (tomtzook) [inline src/main/java/frc/robot/commands/ArmMoveToShooterCommand.java]: again, not relevant and unnecessary

File: src/main/java/frc/robot/subsystems/ArmSystem.java [hunk 7]
```diff
@@ -23,17 +32,33 @@ public ArmSystem() {
 
         motor.restoreFactoryDefaults();
 
+        setUpShuffleboard();
+    }
+
+    private void setUpShuffleboard() {
```
Comment (tomtzook) [inline src/main/java/frc/robot/subsystems/ArmSystem.java]: place `private` method below all the `public` ones

**Review by tomtzook — CHANGES_REQUESTED** [review]
The shuffleboard idea looks good. 

Should replace all the `SmartDashboard` prints though, no?

One glaring problem is your changes of things not necessary or related to the feature you are working on. DON'T. This causes confusion about the changes done here, conflicts with other branches meant to work on that code and some changes are just unnecessary at all.

File: src/main/java/frc/robot/subsystems/ArmSystem.java (hunk 7 above)
Comment (Martysh12) [reply src/main/java/frc/robot/subsystems/ArmSystem.java]: My reasoning for ordering functions that way is that functions closer to other functions are behaviorally similar or in the same category. I'll put it below the `public` methods if that's the code style.

File: src/main/java/frc/robot/subsystems/ArmSystem.java (hunk 4 above)
Comment (Martysh12) [reply src/main/java/frc/robot/subsystems/ArmSystem.java]: Most of this code was written before I heard your rant on the importance of legacy code. I will reverse these changes.

**Comment by Martysh12** [comment]
image

**Comment by Martysh12** [comment]
image

**Comment by Martysh12** [comment]
Added a `ShuffleboardDashboard` class which is responsible for initialising and updating the dashboard tab.
Subsystems must register a supplier like so:
https://github.com/CANBUS-9740/Offseason-2024/blob/2fe0d2e402f688cbdf45cc961ebf11d121e5fe4e/src/main/java/frc/robot/subsystems/DriveSubsystem.java#L69-L72

**Pull request merged**